src/Mod/Sketcher/Gui/ViewProviderSketch.h

```diff
@@ -1,784 +1,800 @@
+ *   Copyright (c) 2009 Juergen Riegel <juergen.riegel@web.de>             *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef SKETCHERGUI_VIEWPROVIDERSKETCH_H
+#define SKETCHERGUI_VIEWPROVIDERSKETCH_H
+
+#include <memory>
+#include <QCoreApplication>
+#include <boost_signals2.hpp>
+#include <Inventor/SoRenderManager.h>
+#include <Inventor/sensors/SoNodeSensor.h>
+
+#include <Base/Parameter.h>
+#include <Base/Placement.h>
+#include <Gui/Document.h>
+#include <Gui/GLPainter.h>
+#include <Gui/Selection.h>
+#include <Mod/Part/Gui/ViewProvider2DObject.h>
+#include <Mod/Part/Gui/ViewProviderGridExtension.h>
+#include <Mod/Part/Gui/ViewProviderAttachExtension.h>
+#include <Mod/Sketcher/App/GeoList.h>
+
+#include "ShortcutListener.h"
+
+
+class TopoDS_Shape;
+class TopoDS_Face;
+class SoSeparator;
+class SbLine;
+class SbVec2f;
+class SbVec3f;
+class SoCoordinate3;
+class SoInfo;
+class SoPointSet;
+class SoTransform;
+class SoLineSet;
+class SoMarkerSet;
+class SoPickedPoint;
+class SoRayPickAction;
+
+class SoImage;
+class QImage;
+class QColor;
+
+class SoText2;
+class SoTranslation;
+class SbString;
+class SbTime;
+
+namespace Part {
+    class Geometry;
+}
+
+namespace Gui {
+    class View3DInventorViewer;
+}
+
+namespace Sketcher {
+    class Constraint;
+    class Sketch;
+    class SketchObject;
+}
+
+namespace SketcherGui {
+
+class EditModeCoinManager;
+class DrawSketchHandler;
+
+enum class SnapMode { // to be moved to SnapManager
+    None,
+    SnapToObject,
+    SnapToAngle,
+    SnapToGrid,
+};
```

Comment:
Shouldn't that be something like:
```suggestion
enum class SnapMode { // to be moved to SnapManager
    None = 0x00,
    SnapToObject = 0x01,
    SnapToAngle = 0x02,
    SnapToGrid = 0x04,
};
```
So that it can be cumulated

Reply:
Good question.

Paddle's current implementation is not. But I am unsure.

I think that one should be answered while dealing with his PR.

It is a little bit beyond just initiatialising as you propose. With a enum class, there is quite some static_casting to be used with binary logic operators, which usually is solved by implementing such operators as free functions. Then direct comparison is not possible... at the end often it is used together with a std::bitset.

We will cross the bridge when we arrive there.

Reply:
Fine for me